src/types.rs

```diff
@@ -183,6 +183,8 @@ pub enum SlaveAddr {
     Default,
     /// Alternative slave address providing bit values for A5, A4, A3, A2, A1 and A0
     Alternative(bool, bool, bool, bool, bool, bool),
+    /// Integer alternative slave address
+    Integer(u8),
```

Comment:
I think it would prefer `Full(u8)` or `AlternativeFull(u8)`

Reply:
either of those is fine, just need some way of deterministically specifying an integer so it can be used from other code. on further thought i think i have a better (but breaking) way to implement this tho

Reply:
What would that way be?

Reply:
shoulda probably made it a second PR but, refactored to rename to `Address` and pass `Into<Address>` where required, then implemented `Default` as well as `From` for `u8` for manual specification and `(bool, bool, ...)` for the boolean version, so you can pass any of em and it, just works (tm).

you have that `check_address` function that does not need to be bound to an instance and could be used to check the address in the constructor if you wanted to return `Result<..>` instead of `Self` there too?

Reply:
Sounds great. Yes, checking the address on `new` as well sounds good. Actually, then the bound could be `TryInto` and the check be done inside the conversion, since currently the address is checked every time except on `new`.

Reply:
added the check to `new`. it's easy to swap to TryInto but, makes the associated bounds more complex so, i would he inclined to leave it as is i think.

Reply:
Sounds good